### ISSUE TITLE
unable to run on RPI0/RPI1 with `arm-linux-musleabi -mcpu=arm1176jzf_s`

### ISSUE THREAD

**Issue by FireFox317** [ziglang/bootstrap#19]
I compiled a binary for people to try on their RPI0/RPI1 (don't have one myself).
@fengb and @shakesoda tried it but got the following error:
```
SIGILL, Illegal instruction. -- 0x025bf470 in __ARMv7ABSLongThunk___libc_start_main ()
```
`Illegal instruction` because something emitted armv7 code, which the RPI0/RPI1 doesn't understand. 

I suspect a few things:
- The target triple `arm-linux-musleabi` to clang/llvm defaults to armv7. `-- LLVM default target triple: arm-linux-musleabi`
- In zigs CMakeLists.txt file we set the cpu to baseline: https://github.com/ziglang/zig/blob/b3aef49eeaa42127ac57ded3b15228db39f806b3/CMakeLists.txt#L436

Edit: @LemonBoy mentioned in IRC: 'lld was tricked into thinking you're targeting a v7 or greater arch'

**Comment by andrewrk** [comment]
good catch, now we need to integrate this build option with the script:
https://github.com/ziglang/zig/commit/d5087ccbc8729a252c6fd0d2fe71a55e4b136e3a

**Comment by andrewrk** [comment]
This might be solved with 9562e24927e0f67835da8a02b00a8033d304b029 - I'm running a clean build to find out.

**Comment by fengb** [comment]
Working on my Pi Zero!
```zig
const std = @import("std");
pub fn main() void {
    std.debug.warn("hello world!\n", .{});
}

// $ ../zig-rpi0-rpi1/zig run blah.zig
// hello world!
```
